src/language/rule/rule.ts

```diff
@@ -89,6 +89,54 @@ export interface IRule {
     applyWithWalker(walker: RuleWalker): RuleFailure[];
 }
 
+export class Replacement {
+    constructor(private start: number, private length: number, private text: string) {
+    }
+
+    public getStart() {
```

Comment:
I'd be fine with making all of these real `get` methods. I find that this sort of method can get frustrating (but fine either way)